app/controllers/api/v2/people_controller.rb

```diff
@@ -13,13 +19,42 @@ def index
 
       PERMITTED_FILTER_PARAMS = %i[police_national_computer criminal_records_office prison_number].freeze
 
+      PERSON_ATTRIBUTES = %i[
+        first_names
+        last_name
+        date_of_birth
+        prison_number
+        criminal_records_office
+        police_national_computer
+        gender_additional_information
+      ].freeze
+      PERMITTED_PERSON_PARAMS = [:type, attributes: PERSON_ATTRIBUTES, relationships: {}].freeze
+
       def filter_params
         params.fetch(:filter, {}).permit(PERMITTED_FILTER_PARAMS).to_h
       end
 
       def supported_relationships
         ::V2::PersonSerializer::SUPPORTED_RELATIONSHIPS
       end
+
+      def person_params
+        params.require(:data).permit(PERMITTED_PERSON_PARAMS).to_h
+      end
+
+      def person_attributes
+        ethnicity_id = params.require(:data).dig(:relationships, :ethnicity, :data, :id)
+        gender_id = params.require(:data).dig(:relationships, :gender, :data, :id)
+
+        # Validate the existence of Ethnicity and Gender
+        Ethnicity.find(ethnicity_id) if ethnicity_id
+        Gender.find(gender_id) if gender_id
+
+        person_params[:attributes].merge(
```

Comment:
To simplify further we can do something like this:
```
person_params[:attributes].merge(
ethnicity: Ethnicity.find(person_params.dig(:relationships, :ethnicity, :data, :id)),
gender: Gender.find(person_params.dig(:relationships, :gender, :data, :id)
)
```
As we will automatically set the ids

Reply:
Was just writing this

Reply:
Or push finding the gender into a method `gender` which includes pulling out the params

```ruby
def ethnicity
  Ethnicity.find(person_params.dig(:relationships, :ethnicity, :data, :id))
end

def gender
  Gender.find(person_params.dig(:relationships, :gender, :data, :id))
end

def person_attributes
  person_params[:attributes].merge(ethnicity: ethnicity, gender: gender)
end
```

Reply:
@cwrw  we need the conditionals because a requirement is that all the params are not mandatory, for now (my understanding is that we could create the person and then later syncs it with nomis).

Reply:
I refactored it with the small ethnicity/gender helper
In the long term, we should probably find a generic solution for this kind of problem.